internal/agent/config.go

```diff
@@ -136,3 +145,15 @@ func (c *Config) validate() error {
 
 	return c.Cloud.validate()
 }
+
+func (c *Config) logDeprecations(logger hclog.Logger) {
```

Comment:
Just for context (other reviewers): I asked why we need this, i.e. can't we just bump with a breaking change release, but @clly explained that we usually log warnings for deprecations before doing that, e.g.  in Consul.

This seems like a safety practice, but maybe we should have a follow-up ticket to perform this cleanup of the `HTTPCollectorEndpoint ` completely to track work?